apps/communications/contacts/js/utilities/tagSelector.js

```diff
@@ -0,0 +1,104 @@
+(function(exports) {
+
+  /* global TAG_OPTIONS, ContactsTag, Contacts, LazyLoader */
+  /* exported TagSelector*/
+
+  'use strict';
+
+  var customTag, customTagReset, tagDone, tagHeader, lazyLoadedTagsDom = false;
+  var contactTag;
+
+  function handleCustomTagReset(ev) {
+    ev.preventDefault();
+    if (customTag) {
+      customTag.value = '';
+    }
+  }
+
+  function handleCustomTag(ev) {
+    if (ev.keyCode === 13) {
+      ev.preventDefault();
+    }
+    ContactsTag.touchCustomTag();
+  }
+
+  function showSelectTag() {
```

Comment:
I would rename this as `showSelectTag(cancelCB)`, so we can remove the implicit dependency with contacts